eigen3/include/ignition/math/eigen3/Util.hh

```diff
@@ -0,0 +1,161 @@
+          cumulants(7) += point(1) * point(2);
+          cumulants(8) += point(2) * point(2);
+        }
+
+        Eigen::Matrix3d covariance;
+
+        cumulants /= static_cast<double>(_vertices.size());
+
+        covariance(0, 0) = cumulants(3) - cumulants(0) * cumulants(0);
+        covariance(1, 1) = cumulants(6) - cumulants(1) * cumulants(1);
+        covariance(2, 2) = cumulants(8) - cumulants(2) * cumulants(2);
+        covariance(0, 1) = cumulants(4) - cumulants(0) * cumulants(1);
+        covariance(1, 0) = covariance(0, 1);
+        covariance(0, 2) = cumulants(5) - cumulants(0) * cumulants(2);
+        covariance(2, 0) = covariance(0, 2);
+        covariance(1, 2) = cumulants(7) - cumulants(1) * cumulants(2);
+        covariance(2, 1) = covariance(1, 2);
+        return covariance;
+      }
+
+      /// \brief Get the oriented 3d bounding box of a set of 3d
+      /// vertices using PCA
+      /// http://codextechnicanum.blogspot.com/2015/04/find-minimum-oriented-bounding-box-of.html
+      /// \param[in] _vertices a vector of 3d vertices
+      /// \return Oriented 3D box
+      inline ignition::math::OrientedBoxd verticesToOrientedBox(
+        const std::vector<math::Vector3d> &_vertices)
+      {
+        math::OrientedBoxd box;
+
+        // Return an empty box if there are no vertices
+        if (_vertices.empty())
+          return box;
+
+        math::Vector3d mean;
+        for (const auto &point : _vertices)
+          mean += point;
+        mean /= static_cast<double>(_vertices.size());
+
+        Eigen::Vector3d centroid = math::eigen3::convert(mean);
+        Eigen::Matrix3d covariance = covarianceMatrix(_vertices);
+
+        // Eigen Vectors
+        Eigen::SelfAdjointEigenSolver<Eigen::Matrix3d>
+          eigenSolver(covariance, Eigen::ComputeEigenvectors);
+        Eigen::Matrix3d eigenVectorsPCA = eigenSolver.eigenvectors();
+
+        // This line is necessary for proper orientation in some cases.
+        // The numbers come out the same without it, but the signs are
+        // different and the box doesn't get correctly oriented in some cases.
+        eigenVectorsPCA.col(2) =
+          eigenVectorsPCA.col(0).cross(eigenVectorsPCA.col(1));
+
+        // Transform the original cloud to the origin where the principal
+        // components correspond to the axes.
+        Eigen::Matrix4d projectionTransform(Eigen::Matrix4d::Identity());
+        projectionTransform.block<3, 3>(0, 0) = eigenVectorsPCA.transpose();
+        projectionTransform.block<3, 1>(0, 3) =
+          -1.0f * (projectionTransform.block<3, 3>(0, 0) * centroid);
+
+        Eigen::Vector3d minPoint(INF_I32, INF_I32, INF_I32);
+        Eigen::Vector3d maxPoint(-INF_I32, -INF_I32, -INF_I32);
+
+        // Get the minimum and maximum points of the transformed cloud.
+        for (const auto &point : _vertices)
+        {
+          Eigen::Vector4d pt(0, 0, 0, 1);
+          pt.head<3>() = convert(point);
```

Comment:
Is this one of the `math::eigen3::convert` methods? If so, we can add the `math::eigen3` namespace so that it's easy for code readers to know which function is being used (adding this namespace would also help avoid potential collisions since `convert` is a generic method name).